compiler/modules/front.syntax/astString.bal

```diff
@@ -244,30 +244,41 @@ function typeDescToWords(Word[] w, TypeDesc td, boolean|BinaryTypeOp wrap = fals
         return;
     }
     else if td is MappingTypeDesc {
-        TypeDesc? rest = td.rest;
+        TypeDesc|INCLUSIVE_RECORD_TYPE_DESC|() rest = td.rest;
         if td.fields.length() > 0 {
-            w.push("record", "{|");
+            w.push("record");
+            if rest != true {
```

Comment:
Swap it round and say `w.push(rest == INCLUSIVE_RECORD_TYPE_DESC ? "{" : "{|");`

Reply:
fixed